### PR TITLE
feat(language): add configurable extensions

### PR DESCRIPTION
**Pull request by lnu**
### Prerequisites

- [x] I have read and understood the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes/features)

### Description

[Description of the change]

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: src/segment_language.go [hunk 1]
```diff
@@ -151,6 +154,12 @@ func (l *language) string() string {
 }
 
 func (l *language) enabled() bool {
+	// override default extensions if needed
+	ext := l.props.getString(LanguageExtensions, "")
```
Comment (JanDeDobbeleer) [inline src/segment_language.go]: string or array?

Comment (lnu) [reply src/segment_language.go]: `getStringArray` added

**Comment by lnu** [comment]
the block 
```
 "go.languageServerExperimentalFeatures": {
    "diagnostics": false
  },
```
was causing some issues when loading the go extensions

**Review by JanDeDobbeleer — APPROVED** [review]

File: src/segment_language.go [hunk 2]
```diff
@@ -98,6 +99,8 @@ const (
 	EnableVersionMismatch Property = "enable_version_mismatch"
 	// HomeEnabled displays the segment in the HOME folder or not
 	HomeEnabled Property = "home_enabled"
+	// LanguageExtensions the list of extensions to validate
+	LanguageExtensions Property = "extensions"
```
Comment (JanDeDobbeleer) [inline src/segment_language.go:103]: silent launch? Let's see if it works according to expectations :-)

Comment (lnu) [reply src/segment_language.go:103]: Power user only 😅

**Pull request merged**